lib/mayaHydra/mayaPlugin/renderOverride.cpp

```diff
@@ -2941,7 +2943,8 @@ void MtohRenderOverride::_SetRenderPurposeTags(const MayaHydraParams& delegatePa
         }
     }
 #endif
-}
 
+    _purposeFilteringSceneIndex->UpdatePrimsFromIncludedPurposes(RenderGlobalsUtils::GetIncludedPurposes());
```

Comment:
Gave up on trying to set included purposes only when they change, intricate code that goes through a MEL command, was unreliable and complex.  Now simply setting them on every draw.